### ISSUE TITLE
The webphone have whisper, listen, barge, break features?

### ISSUE THREAD

**Issue by beatific-angel** [InnovateAsterisk/Browser-Phone#422]
webphone is very nice to using.
one question
The webphone have whisper, listen, barge, break features?

**Comment by InnovateAsterisk** [comment]
Typically these kinds of features are done with feature codes on you Asterisk Dialplan. The Browser phone can be used to dial such feature code, and listen to a conversation, just like a normal handset could.

**Comment by beatific-angel** [comment]
so those features are not in here, but you can do this ?
right?

**Comment by beatific-angel** [comment]
I can make dialplan for whisper, barge and listen.
but I dont know how to connect the dialplan with webphone.

this is dialplan for whisper, barge and listen.
[ext-local-custom]

;listen
exten => _*222x.#,1,Macro(user-callerid,)
exten => _*222x.#,n,Answer
exten => _*222x.#,n,NoCDR
exten => _*222x.#,n,Wait(1)
exten => _*222x.#,n,ChanSpy(sip/${EXTEN:4},q)
exten => _*222x.#,n,Hangup

;whisper
exten => _*223x.#,1,Macro(user-callerid,)
exten => _*223x.#,n,Answer
exten => _*223x.#,n,NoCDR
exten => _*223x.#,n,Wait(1)
exten => _*223x.#,n,ChanSpy(sip/${EXTEN:4},qw)
exten => _*223x.#,n,Hangup

;barge
exten => _*224x.#,1,Macro(user-callerid,)
exten => _*224x.#,n,Answer
exten => _*224x.#,n,NoCDR
exten => _*224x.#,n,Wait(1)
exten => _*224x.#,n,ChanSpy(SIP/${EXTEN:4},qB)
exten => _*224x.#,n,Hangup

**Comment by InnovateAsterisk** [comment]
This really is an Asterisk Dial-plan question. Based on the dial-plan provided, it looks like your feature codes are  ```*222``` for listen, ```*223``` for whisper and ```*224``` for barge. 
The way to use them is to simply dial *222 followed by the extension you want to listen to, so ```*222100``` would listen to extension 100, and all the combinations etc.

**Comment by beatific-angel** [comment]
ofc , above my dialplan is asterisk dial plan.
but please understand my question.
My mean, how to connect call to the ongoing call in web phone?

please check this link.
https://d1qcxhd2x8x3pn.cloudfront.net/dashboards-drive-1588323311.mp4

In the video, if I request call, it will connect to the ongoing call.
Maybe my explanation is not good. Anyway please check the video.

**Comment by InnovateAsterisk** [comment]
> In the video, if I request call, it will connect to the ongoing call.

No, this is what I looks like, but whats happening behind the scene, is that the dialer client they provide is making a simple call just like any other call, except that the number its calling is a special feature code, and by dialing the feature code, your call is "connected to the ongoing call". 

As i said before, just dial the number from the Browser Phone ```*222100```. This will "connect" your browser phone to the ongoing call that the person on extension 100 is on. If you want to connect to extension 200, then dial *222200, and so on.

**Comment by beatific-angel** [comment]
really? ok. will try soon.

**Comment by InnovateAsterisk** [comment]
Asterisk is a B2BUA (Back to Back User Agent), this means all signalling and media all goes "through" the server. It allows this like this to be possible. So long as you don't re-invite, or use direct media.

**Comment by beatific-angel** [comment]
I tried. 
via asterisk log, I can see chanspy work. 
but I can not listen any voices in the spy channel and I can not say to anyone.
why these problems occurred?
I am not sure.
volume or other problem?

**Comment by InnovateAsterisk** [comment]
First make sure a normal audio call works between two webrtc users. Then make sure the spy features work "from" a regular desk phone, or non-webrtc phone (as in the device you are spying with).

Confirmed these two things work.

**Comment by beatific-angel** [comment]
just I checked,
web rtc phone vs web rtc phone call works correctly.
but I can not use whisper and barge feature too.

**Comment by beatific-angel** [comment]
I tested web phone and soft phone call.
in this case, I can hear audio. but if I try to speak, I can hear strange changed voice.

**Comment by beatific-angel** [comment]
and soft phone and outbound call tested. in this case, i can hear audio in the channel.
so i think this problem is related to webrtc.

**Comment by InnovateAsterisk** [comment]
Can you try using only ulaw or alaw as the codec only on all extensions including the webrtc ones.

**Comment by beatific-angel** [comment]
![image](https://user-images.githubusercontent.com/64799935/217634210-abb626a9-94f3-4c00-a9f6-6ea0c229e61e.png)
 I am already using the ulaw, and alaw .
